blockmanager.go

```diff
@@ -524,7 +555,7 @@ func (b *blockManager) current() bool {
 	// TODO(oga) we can get chain to return the height of each block when we
 	// parse an orphan, which would allow us to update the height of peers
 	// from what it was at initial handshake.
-	if err != nil || height < int64(b.syncPeer.lastBlock) {
+	if err != nil || height < int64(b.syncPeer.LastBlock()) {
```

Comment:
Ah right.  Well this check needs to compare against the latest announced height to be correct.  Unless I've missed something with your new code, the `LastBlock()` is not updated after the initial connection.  This check is for selecting a new sync peer and therefore needs to be comparing against the latest known height as ascertained from their inventory announcements.